src/library/scala/collection/immutable/NumericRange.scala

```diff
@@ -402,7 +402,7 @@ object NumericRange {
       val startside = num.sign(start)
       val endside = num.sign(end)
       num.toInt{
-        if (num.gteq(num.times(startside, endside), zero)) {
+        if (num.gt(num.times(startside, endside), zero)) {
```

Comment:
`[0, Long.MinValue]` by any negative step is handled here, while it's definitely not safe to subtract 0 with MinValue.

Reply:
The condition is now a very slow way to write `startside == endside`.  So one could use that instead.  But the reason to include zero is to make the most usual range, which is [0, n), to be easy to calculate.

Can we just drop `startside` and `endside` entirely and instead compute  `num.lt(start, zero) == num.lt(end, zero)` (i.e. either both negative or both non-negative)?  This will only admit Int.MinValue when the range is no bigger than -1 to Int.MinValue, which is okay to subtract.